### DISCUSSION TITLE
Rename refactoring is here

### DISCUSSION THREAD

**Discussion by WaynePhillipsEA** [twinbasic/twinbasic#253] [General]
With twinBASIC v0.10.3333, we've now got rename refactoring available.  To activate it, simply put the caret on a symbol, and press the F2 key.    Disclaimer;  It's not perfect yet.   There are some edge cases known to be not yet working, namely generics, named arguments and default values, and probably a few more for you to find :)

![tb_renameRefactoring1](https://user-images.githubusercontent.com/25738109/124143472-0dadd880-da83-11eb-882f-4c3598da51d8.gif)

**Comment by Kr00l** [comment]
I think that I found a bug related to API declares.

When I go to the API call in the code I can rename it.
![image](https://user-images.githubusercontent.com/33756666/124149527-f37af700-da90-11eb-8278-f6d7032d4a12.png)

However, when I am on the API declare itself then renaming is not allowed. But technically the first scenario is the same as this scenario.
![image](https://user-images.githubusercontent.com/33756666/124149654-15747980-da91-11eb-9f9b-603ba264b837.png)

**Reply by WaynePhillipsEA** [reply]
Yep you did!  Thanks

**Reply by WaynePhillipsEA** [reply]
Fixed in v0.10.3412, thanks!
